instrumentation/events/lttng-module/net.h

```diff
@@ -49,6 +230,36 @@ LTTNG_TRACEPOINT_EVENT_CLASS(net_dev_template,
 		ctf_integer_hex(void *, skbaddr, skb)
 		ctf_integer(unsigned int, len, skb->len)
 		ctf_string(name, skb->dev->name)
+		ctf_enum(net_network_header, unsigned char,
+			network_header_type, __get_network_header_type(skb))
+		ctf_custom_field(
+			ctf_custom_type(
+				.atype = atype_variant,
+				.u.variant.tag_name = "network_header_type",
+				.u.variant.choices = network_fields,
+				.u.variant.nr_choices = ARRAY_SIZE(network_fields),
+			),
+			network_header,
+			ctf_custom_code(
+				switch (__get_network_header_type(skb)) {
+				case NH_IPV4: {
+					ctf_array_type(uint8_t, ip_hdr(skb),
+							sizeof(struct iphdr))
+					ctf_align(uint8_t)
```

Comment:
----- On Jul 13, 2016, at 1:26 PM, Geneviève Bastien <notifications@github.com> wrote: 

> In instrumentation/events/lttng-module/net.h :
> > +		ctf_enum(net_network_header, unsigned char,
> > +			network_header_type, __get_network_header_type(skb))
> > +		ctf_custom_field(
> > +			ctf_custom_type(
> > +				.atype = atype_variant,
> > +				.u.variant.tag_name = "network_header_type",
> > +				.u.variant.choices = network_fields,
> > +				.u.variant.nr_choices = ARRAY_SIZE(network_fields),
> > +			),
> > +			network_header,
> > +			ctf_custom_code(
> > +				switch (__get_network_header_type(skb)) {
> > +				case NH_IPV4: {
> > +					ctf_array_type(uint8_t, ip_hdr(skb),
> > +							sizeof(struct iphdr))
> > +					ctf_align(uint8_t)

> So here is where I should align at 16 instead of 8
You should align on 16 before the ctf_array_type if the ctf_array_type contains 
fields that are 16-bit aligned. 

Mathieu 

> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub , or mute the thread .

-- 
Mathieu Desnoyers 
EfficiOS Inc. 
http://www.efficios.com